dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/AbstractInterfaceConfig.java

```diff
@@ -689,7 +689,10 @@ public ApplicationConfig getApplication() {
     }
 
     public void setApplication(ApplicationConfig application) {
-        ConfigManager.getInstance().setApplication(application);
+        if (!ConfigManager.getInstance().getApplication().isPresent() ||
+                !ConfigManager.getInstance().getApplication().get().equals(application)) {
+            ConfigManager.getInstance().setApplication(application);
+        }
```

Comment:
Why should we need this logic? The logic, including the null check and duplicate check have  implemented in org.apache.dubbo.config.context.ConfigManager#setApplication.  Why should be added it again?

So as the other change in this file.

Reply:
such as AbstractInterfaceConfig#createApplicationIfAbsent , if `this.application` is null and the `ApplicationConfig` in `ConfigManager`  is not null , it will reset   `ApplicationConfig` variable，that always causes an error in `ConfigManager#checkDuplicate`

Reply:
Still do not quite get it. in AbstractInterfaceConfig#createApplicationIfAbsent , if `this.application` is null, it will directly return.
Can you write a unit test to reproduce the error?

Reply:
I thought about it again, this code seems to be doing this deliberately. my logic is duplicate check and block the IllegalStateException, i should remove them . 

And  because `ConfigManager#checkDuplicate` is logically wrong, it will cause some unit test errors which is  the purpose of changing this logic at the beginning...